File: lib/java/opentoken/src/main/java/com/truveta/opentoken/attributes/validation/AgeRangeValidator.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.attributes.validation;

import lombok.NoArgsConstructor;

/**
 * A Validator that asserts that the age value is within
 * an acceptable range (between 0 and 120).
 * 
 * This validator checks that ages are:
 * - Not negative
 * - Not greater than 120 years
 * 
 * If the age is outside this range, the validation fails.
 */
@NoArgsConstructor
public final class AgeRangeValidator implements SerializableAttributeValidator {

    private static final long serialVersionUID = 1L;

    private static final int MIN_AGE = 0;
    private static final int MAX_AGE = 120;

    /**
     * Validates that the age value is within acceptable range.
     * 
     * @param value the age string to validate
     * @return true if the age is within acceptable range (0 to 120),
     *         false otherwise
     */
    @Override
    public boolean eval(String value) {
        if (value == null || value.trim().isEmpty()) {
            return false;
        }

        try {
            int age = Integer.parseInt(value.trim());
            return age >= MIN_AGE && age <= MAX_AGE;
        } catch (NumberFormatException e) {
            // If the value cannot be parsed as an integer, it's invalid
            return false;
        }
    }
}

File: lib/python/opentoken/src/main/opentoken/attributes/validation/age_range_validator.py
"""
Copyright (c) Truveta. All rights reserved.
"""

from opentoken.attributes.validation.serializable_attribute_validator import SerializableAttributeValidator


class AgeRangeValidator(SerializableAttributeValidator):
    """
    A validator that asserts that an age value is within an acceptable range (0-120).

    This validator checks that ages are:
    - Not less than 0
    - Not greater than 120

    If the age is outside this range, the validation fails.
    """

    MIN_AGE = 0
    MAX_AGE = 120

    def __init__(self):
        """Initialize the validator with default age range (0-120)."""
        pass

    def eval(self, value: str) -> bool:
        """
        Validate that the age value is within acceptable range.

        Args:
            value: The age string to validate

        Returns:
            True if the age is within acceptable range (0-120), False otherwise
        """
        if not value or not value.strip():
            return False

        try:
            age = int(value.strip())
            return self.MIN_AGE <= age <= self.MAX_AGE
        except ValueError:
            return False
